backends-velox/src/test/scala/io/glutenproject/execution/VeloxMetricsSuite.scala

```diff
@@ -0,0 +1,92 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.glutenproject.execution
+
+import io.glutenproject.GlutenConfig
+
+import org.apache.spark.sql.execution.adaptive.AdaptiveSparkPlanHelper
+import org.apache.spark.sql.internal.SQLConf
+
+class VeloxMetricsSuite extends VeloxWholeStageTransformerSuite with AdaptiveSparkPlanHelper {
+  override protected val backend: String = "velox"
+  override protected val resourcePath: String = "/tpch-data-parquet-velox"
+  override protected val fileFormat: String = "parquet"
+
+  override def beforeAll(): Unit = {
+    super.beforeAll()
+
+    spark
+      .range(100)
+      .selectExpr("id as c1", "id % 3 as c2")
+      .write
+      .format("parquet")
+      .saveAsTable("metrics_t1")
+
+    spark
+      .range(200)
+      .selectExpr("id as c1", "id % 3 as c2")
+      .write
+      .format("parquet")
+      .saveAsTable("metrics_t2")
+  }
+
+  override protected def afterAll(): Unit = {
+    spark.sql("drop table metrics_t1")
+    spark.sql("drop table metrics_t2")
+
+    super.afterAll()
+  }
+
+  test("test sort merge join metrics") {
```

Comment:
@ulysses-you Do we need to add the test with post project? And also add the hash join test for both with pre project and post project?

Reply:
Join always has the post project. The numOutputRows/numOutputVectors metrics are from post project.